test/test-span-data.ts

```diff
@@ -14,177 +14,174 @@
  * limitations under the License.
  */
 
-'use strict';
+import * as assert from 'assert';
 
-import './override-gcp-metadata';
-import { cls } from '../src/cls';
-import { Constants } from '../src/constants';
+import {Constants, SpanDataType} from '../src/constants';
+import {BaseSpanData, ChildSpanData, RootSpanData} from '../src/span-data';
+import {Trace} from '../src/trace';
 import {TraceLabels} from '../src/trace-labels';
-import { RootSpanData, ChildSpanData } from '../src/span-data';
-import { Trace } from '../src/trace';
-import { traceWriter } from '../src/trace-writer';
-
-var assert = require('assert');
-var common = require('./plugins/common'/*.js*/);
-
-function createRootSpanData(name, traceId?, parentId?, skipFrames?) {
-  return new RootSpanData(
-    { projectId: '', traceId: traceId, spans: [] },
-    name,
-    parentId,
-    skipFrames
-  );
-}
-
-describe('SpanData', function() {
-  var agent;
-  before(function() {
-    agent = require('../..').start({
-      projectId: '0',
-      samplingRate: 0
-    });
+import {traceWriter, TraceWriter, TraceWriterConfig} from '../src/trace-writer';
+
+import {TestLogger} from './logger';
+import * as traceAgentModule from './trace';
+
+describe('SpanData', () => {
+  class CaptureSpanTraceWriter extends TraceWriter {
+    writeSpan(trace: Trace) {
+      capturedTrace = trace;
```

Comment:
Should this also have an assertion that verifies `capturedTrace` is only set once?

Reply:
Done